week_6/testing/src/app/header/header.component.spec.ts

```diff
@@ -0,0 +1,37 @@
+import { HeaderComponent } from './header.component';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { QuizService } from '../data/quiz.service';
+import { StorageService } from '../storage.service';
+import { By } from '@angular/platform-browser';
+
+describe('HeaderComponent', () => {
+  let component: HeaderComponent;
+  let fixture: ComponentFixture<HeaderComponent>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HeaderComponent],
+      providers: [QuizService, StorageService],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(HeaderComponent);
+    component = fixture.componentInstance;
+
+    fixture.detectChanges();
+  });
+
+  it('renders correctly', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Switches between light and dark mode', () => {
```

Comment:
start a description with small letter. use 'should switch between light and dark mode'

Reply:
done